test/buffer_manager_test_utils.h

```diff
@@ -3,14 +3,17 @@
 #include "test_utils.h"
 
 namespace celerity {
+
 namespace detail {
+	struct accessor_testspy {
```

Comment:
I think moving it once we need it is fine, in particular since I don't think we'll need this for anything other than testing the buffer manager. Usually we either use mock buffers without accessors, or we do a full integration-style test case where we construct accessors using their public APIs.